init.lua

```diff
@@ -82,6 +105,7 @@ local function preview()
 
    -- create a list that holds the clients to preview, from left to right
    local leftRightTab = {}
+   local leftRightTabToAltTabIndex = {} -- save mapping from leftRightTab to altTabTable as well
```

Comment:
Overall the codebase is already on the way to "unmaintainable", that's why I am a little bit harsh to feature additions and especially new state holders. A major refactoring is on my list but not as high on my priority list as other projects. I am looking forward to see some improvements in your fork. Looks promising!

The referenced code block was
```lua
preview_widgets[i]:connect_signal("mouse::enter", function()
         altTabIndex = cycle(altTabTable, altTabIndex, leftRightTabToAltTabIndex[i] - altTabIndex)
end)
```
but using the index here alone is not possible as you already mentioned above. 

For this one I would say that the user does more profit from the feature than from cleaner code.
Nevertheless, thank you for taking time to explain your motivation.